### PR TITLE
solvestack coding challenge

### PR DESCRIPTION
**Pull request by CactusWren2020**
I wasn't sure if the /v1/stacks REST should require authentication, so I left in the jwtConfig.isAuthenticated middleware. It works without authentication if the call is removed.

### CODE REVIEW

**Comment by CactusWren2020** [comment]
Added tests for stacks api and removed NODE_ENV=TEST to get test working.

File: src/components/User/validation.ts [hunk 1]
```diff
@@ -35,6 +35,25 @@ class UserValidation extends Validation {
...
+    //         name: Joi.string().required(),
+    //         email: Joi.string().email({
+    //             minDomainAtoms: 2
+    //         }).required()
+    //     });
+
+    //     return Joi.validate(params, schema);
+    // }
+
+
```
Comment (tomboolean) [inline src/components/User/validation.ts:56]: nit - commented code

File: test/api.js [hunk 2]
```diff
@@ -35,6 +46,22 @@ describe('API', () => {
...
+        request(app)
+            .post('/v1/stacks')
+            .send(newStack)
+            .set('x-access-token', global.token)
+            .expect((res) => {
+                res.status.should.equal(201);
+                res.body.should.have.property('name');
+            })
+            .end(done);
+    });
```
Comment (tomboolean) [inline test/api.js:64]: Neat! Appreciate you taking the time to make tests as well.

**Review by tomboolean — APPROVED** [review]
Excellent job! You created POST, GET by id and GET list methods and added happy-path tests for the list and create.

**Pull request merged**